homeassistant/components/intellifire/config_flow.py

```diff
@@ -127,3 +131,35 @@ async def async_step_user(
             return await self.async_step_pick_device()
         LOGGER.debug("Running Step: manual_device_entry")
         return await self.async_step_manual_device_entry()
+
+    async def async_step_dhcp(self, discovery_info: DhcpServiceInfo) -> FlowResult:
+        """Handle DHCP Discovery."""
+
+        # Run validation logic on ip
+        host = discovery_info.ip
+        self._async_abort_entries_match({CONF_HOST: host})
+        try:
+            serial = await validate_host_input(host)
+        except (ConnectionError, ClientConnectionError):
+            return self.async_abort(reason="not_intellifire_device")
+
+        self._discovered_hosts.append((host, serial))
+        return await self.async_step_dhcp_confirm()
+
+    async def async_step_dhcp_confirm(self, user_input=None):
+        """Attempt to confirm."""
+
+        if user_input is None:
+            # Show the confirmation dialog
+            return self.async_show_form(step_id="dhcp_confirm")
+
+        print("USER INPUT: ", user_input)
+        # Add the hosts one by one
+        for host, serial in self._discovered_hosts:
+
+            await self.async_set_unique_id(serial)
+            self._abort_if_unique_id_configured(updates={CONF_HOST: host})
+            return self.async_create_entry(
+                title=f"Fireplace {serial}",
+                data={CONF_HOST: host},
+            )
```

Comment:
This is going to setup the first discovered one (you can only return once). We should only setup the one that was discovered.

Reply:
I _think_ I follow but you may need to look this one over again. I took a look at what they did in flux_led 

<img width="559" alt="image" src="https://user-images.githubusercontent.com/6491743/158288673-2e94e9cb-ab48-48b9-ac97-43705a4582b9.png">

- nothing broke but I don't see anything different
also went with:

```json
    "flow_title": "Discovered Fireplace ({host})",
```

For the placeholder -> from reading the docs its supposed to only render I guess if the host thing is set?